File: cpp/Barang.cpp
#include <string>

class Barang
{
private:
    int id;
    std::string nama;
    std::string kategori;
    // using string because some no part contain letter
    std::string no_part;
    std::string manufaktur;
    int harga;

public:
    // Constructor tanpa parameter
    Barang()
    {
        id = 0;
        nama = "";
        kategori = "";
        no_part = "";
        manufaktur = "";
        harga = 0;
    };

    // Constructor dengan parameter
    Barang(int id, std::string nama, std::string kategori, std::string no_part, std::string manufaktur, int harga)
    {
        //use this-> to specify it refer to "id" from this class(Barang) not "id" from parameter
        this->id = id;
        this->nama = nama;
        this->kategori = kategori;
        this->no_part = no_part;
        this->manufaktur = manufaktur;
        this->harga = harga;
    };

    // Getter and Setter for id
    int getId() const { return id; }
    void setId(int value) { id = value; }

    // Getter and Setter for nama
    std::string getNama() const { return nama; }
    void setNama(const std::string& value) { nama = value; }

    // Getter and Setter for kategori
    std::string getKategori() const { return kategori; }
    void setKategori(const std::string& value) { kategori = value; }

    // Getter and Setter for no_part
    std::string getNoPart() const { return no_part; }
    void setNoPart(const std::string& value) { no_part = value; }

    // Getter and Setter for manufaktur
    std::string getManufaktur() const { return manufaktur; }
    void setManufaktur(const std::string& value) { manufaktur = value; }

    // Getter and Setter for harga
    int getHarga() const { return harga; }
    void setHarga(int value) { harga = value; }

    ~Barang(){

    };
};

File: Folder Java/Barang.java
import java.util.*;

public class Barang {
    private int id;
    private String nama;
    private String kategori;
    private String noPart;
    private String manufaktur;
    private int harga;

    // Constructor tanpa parameter
    public Barang() {
        id = 0;
        nama = "";
        kategori = "";
        noPart = "";
        manufaktur = "";
        harga = 0;
    }

    // Constructor dengan parameter
    public Barang(int id, String nama, String kategori,
                  String noPart, String manufaktur, int harga) {
        this.id = id;
        this.nama = nama;
        this.kategori = kategori;
        this.noPart = noPart;
        this.manufaktur = manufaktur;
        this.harga = harga;
    }

    // Getter & Setter
    public int getId() { return id; }
    public void setId(int value) { id = value; }

    public String getNama() { return nama; }
    public void setNama(String value) { nama = value; }

    public String getKategori() { return kategori; }
    public void setKategori(String value) { kategori = value; }

    public String getNoPart() { return noPart; }
    public void setNoPart(String value) { noPart = value; }

    public String getManufaktur() { return manufaktur; }
    public void setManufaktur(String value) { manufaktur = value; }

    public int getHarga() { return harga; }
    public void setHarga(int value) { harga = value; }
}
